File: app/src/main/java/com/itba/runningMate/repository/achievements/AchievementsStorage.java
package com.itba.runningMate.repository.achievements;

public interface AchievementsStorage {

    String ACHIEVEMENTS_PREFERENCES_FILE = "PREF_RUNNING_MATE_ACHIEVEMENTS";

    void persistState();

    double getTotalDistance();

    void decreaseTotalDistance(double distance);

    void increaseTotalDistance(double distance);

}

File: app/src/main/java/com/itba/runningMate/repository/achievements/AchievementsStorage.kt
package com.itba.runningMate.repository.achievements

interface AchievementsStorage {

    fun persistState()

    fun getTotalDistance(): Double

    fun decreaseTotalDistance(distance: Double)

    fun increaseTotalDistance(distance: Double)

    companion object {
        const val ACHIEVEMENTS_PREFERENCES_FILE = "PREF_RUNNING_MATE_ACHIEVEMENTS"
    }
}